docs/tutorial/flink.md

```diff
@@ -1,94 +1,407 @@
+
+### Where Data is Discovered
+
+Recipe input is defined in the recipe itself. Most recipes will define a `pangeo_forge_recipes.patterns.FilePattern` that provides the pipeline with input file locations.
+The example below taken from the [integration test recipe](https://github.com/pforgetest/gpcp-from-gcs-feedstock/blob/main/feedstock/recipe.py)
+
+   ```python
+   import apache_beam as beam
+   import pandas as pd
+
+   from pangeo_forge_recipes.patterns import ConcatDim, FilePattern
+   from pangeo_forge_recipes.transforms import OpenURLWithFSSpec, OpenWithXarray, StoreToZarr
+   
+   dates = [
+       d.to_pydatetime().strftime('%Y%m%d')
+       for d in pd.date_range("1996-10-01", "1999-02-01", freq="D")
+   ]
+
+   def make_url(time):
+       url_base = "https://storage.googleapis.com/pforge-test-data"
+       return f"{url_base}/gpcp/v01r03_daily_d{time}.nc"
+
+   concat_dim = ConcatDim("time", dates, nitems_per_file=1)
+   pattern = FilePattern(make_url, concat_dim)
+
+   recipe = (
+       beam.Create(pattern.items())
+       | OpenURLWithFSSpec()
+       | OpenWithXarray(file_type=pattern.file_type, xarray_open_kwargs={"decode_coords": "all"})
+       | StoreToZarr(
+           store_name="gpcp",
+           combine_dims=pattern.combine_dim_keys,
+       )
+   )
+   ```
+
+### Where Data is Cached
+
+The inputs and metadata can be cached so items don't have to be resolved again during the next recipe run. Defining where
+the caching happens is part of the runner file configuration. More about that in the [next session](#configfiles). Note 
+that in those upcoming examples keys `InputCacheStorage`` target these options.
+
+### Where Data is Output
+
+Defining where the data is output is also part of the runner file configuration. More about that in the [next session](#configfiles). Note 
+that in those upcoming examples the key `TargetStorage` defines this option.
+
+
+<a name="configfiles"></a>
+
+### Configuration Files
+
+Let's build a configuration file for where data is cached and output. Below we show two different formats for this configuration file.
+
+Notice the use of `{{job_name}}` in the `root_path` configuration examples below.
+`{{job_name}}` is special within `root_path` and will be treated as a template-value based on
+`Bake.job_name` which can be provided through the CLI or in the configuration file itself and, failing
+to be provided, will be generated automatically.
+
+Also notice that we are going to store everything in s3 below because we chose `s3fs.S3FileSystem`. This isn't a requirement.
+Pangeo-Forge aims to be storage agnostic. By depending on `fsspec`  we're able to plug in supported backends.
+Review other well-known `fsspec` [built-in implementations](https://filesystem-spec.readthedocs.io/en/latest/api.html#built-in-implementations)
+and `fsspec` [third-party implementations](https://filesystem-spec.readthedocs.io/en/latest/api.html#other-known-implementations).
+
+1. JSON configuration:
```

Comment:
Reference here that the [traitlets](https://traitlets.readthedocs.io/en/stable/) library is used for config, and that the following config files are equivalent (I see it's mentioned a few paragraphs up, but should be closer down here). Also consider just picking one format and sticking to it (I suggest python, since users can't actually copy paste this JSON - it's invalid due to the presence of comments!) rather than providing two equivalents. We can just link out to the fact that it can be json too.

Reply:
Added the ref to the traitlets library. 

Since the JSON here is just mentioned in passing as another alternative I left it in but removed the comments from the JSON so it might be usable if copied. The remaining examples use Python traitlet configuration so preference is intuited given the examples. I think that works